flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java

```diff
@@ -97,6 +97,31 @@ public void clean() {
     super.clean();
   }
 
+  private void insertRows(String partition, String branch, Table table, Row... rows)
+      throws IOException {
+    GenericAppenderHelper appender = new GenericAppenderHelper(table, FORMAT, temporaryDirectory);
+
+    GenericRecord gRecord = GenericRecord.create(table.schema());
+    List<Record> records = Lists.newArrayList();
+    for (Row row : rows) {
+      records.add(
+          gRecord.copy(
+              "id", row.getField(0),
+              "data", row.getField(1),
+              "dt", row.getField(2)));
+    }
+
+    if (partition != null) {
+      appender.appendToTable(TestHelpers.Row.of(partition, 0), records);
```

Comment:
we could also append to a table with both partition and branch set

Reply:
see if `appendToTable` can take the `MAIN_BRANCH`. then we can set the branch name outside this if-else block with partition.